### ISSUE TITLE
Proposal: NavigationView control should be updated to support swipe.

### ISSUE THREAD

**Issue by shaheedmalik** [microsoft/microsoft-ui-xaml#3123]
# Proposal: NavigationView control should be updated to support swipe.

## Summary
The NavigationView control should be updated to support swipe gestures to change to different menus and items.

## Scope

| Capability | Priority |
| :---------- | :------- |
| Improve the usability of touch applications | Must |
| This proposal will allow end users to accomplish X | Should |
| Provide a way Win32 apps that use Ribbon more touch friendly | Could |
| This proposal will also solve Proposal #848 | Would |

## Important Notes
As an touch user, that uses even his non touch desktop in Tablet Mode, I hate tapping on things I should be able to swipe. I would love to swipe between menus to find what I want to faster.

Here's a proof of concept already in a WinUI application.
![Nightingale-REST-Client-2020-08](https://user-images.githubusercontent.com/51347950/90197940-39a66980-dd9e-11ea-960f-c5244e35dd61.gif)
@dpaulino did a great job on this feature implementing it into the Nightingale Rest client. I feel it is something that all WinUI apps should take advantage of. From using it, it also solves the problem of menus not following the finger of controls such as Pivot. Since Pivot has been put in "maintenance mode" , let's improve the usability of NavigationView.

**Comment by mdtauk** [comment]
That is a Pivot, and the Pivot supports swiping.

**Comment by shaheedmalik** [comment]
> That is a Pivot, and the Pivot supports swiping.
Correct me if I am wrong, but pivot didn't support following of the finger. This particular customized control, follows the person's finger. Unless the creator, added finger following, and styled it like the NavigationView, I am pretty sure that is NavigationView.

**Comment by dpaulino** [comment]
@shaheedmalik it's actually using pivot 😛

**Comment by mdtauk** [comment]
> > That is a Pivot, and the Pivot supports swiping.
> 
> Correct me if I am wrong, but pivot didn't support following of the finger. This particular customized control, follows the person's finger. Unless the creator, added finger following, and styled it like the NavigationView, I am pretty sure app is using NavigationView instead of Pivot.

Pivot does follow your finger, and when it moves a certain distance, and releases, it completes the transition to the next Pivot Item.

It is only the selection indicator that does not follow your finger, or animate it's position.  The team felt there was not enough benefit to updating this control to support this, like the NavigationView's selection indicator. #2585

**Comment by shaheedmalik** [comment]
> > > That is a Pivot, and the Pivot supports swiping.
> > 
> > 
> > Correct me if I am wrong, but pivot didn't support following of the finger. This particular customized control, follows the person's finger. Unless the creator, added finger following, and styled it like the NavigationView, I am pretty sure app is using NavigationView instead of Pivot.
> 
> Pivot does follow your finger, and when it moves a certain distance, and releases, it completes the transition to the next Pivot Item.
> 
> It is only the selection indicator that does not follow your finger, or animate it's position. The team felt there was not enough benefit to updating this control to support this, like the NavigationView's selection indicator. #2585

Right. I commented on that one. NavigationView should really be updated to pivot.

**Comment by winston-de** [comment]
> As an touch user, that uses even his non touch desktop in Tablet Mode, I hate tapping on things I should be able to swipe. I would love to swipe between menus to find what I want to faster.

I would like to add that pivots are great with trackpads, too. I love being able to use two-finger horizontal swipe to  quickly navigate through pages without every having to click or tap.

**Comment by mdtauk** [comment]
> Right. I commented on that one. NavigationView should really be updated to pivot.
>

I disagree with this, the Pivot and NavigationView (set to top mode) are fundamentally different controls, and swiping with NavigationView would have lots of usability issues with the content in the pages themselves, if it "ate" the input.

Also NavigationView pages, do not necessarily have a chronological or logical grouping, so quick swipe actions would have to go through each of the pages one by one, which will add to load times, and could cause confusion.

Just because the top orientated NavigationView has a similar appearance to the Pivot - doesn't mean the swipe gesture makes logical sense.

**Comment by shaheedmalik** [comment]
Now that the Pivot control has been removed from Project Reunion and WinUI 3, I believe this point should be addressed.

#4480
